### PR TITLE
Fix uniform ref implementation

### LINKED ISSUE #1954: RuntimeError: Failed running call_function aten.uniform(...
### 🐛 Describe the bug

```
import torch
import torch._dynamo as dynamo

@dynamo.optimize()
def forward(x):
    return torch.ops.aten.uniform(x)

forward(torch.rand(1000))
```

Error
```
RuntimeError: Failed running call_function aten.uniform(*(FakeTensor(FakeTensor(..., device='meta', size=(1000,)), cpu),), **{}):
uniform() missing 2 required keyword-only arguments: 'dtype' and 'device'
```

### Error logs

Ful log: https://gist.github.com/silvasean/c502b2a4df4feb97403cdbd8b56eb1b5

### Minified repro

See above

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes https://github.com/pytorch/torchdynamo/issues/1954

### CODE REVIEW

**Comment by yanboliang** [comment]
I'm pretty new to this component, not sure what I'm doing. @ezyang @ngimel Can you coach me if this is the right direction? Where I need to add unit tests?

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -4687,21 +4687,25 @@ def scalar_tensor(
 
 @register_decomposition(torch.ops.aten.uniform)
 def uniform(
-    shape: ShapeType,
+    x: TensorLikeType,
```
Comment (yanboliang) [inline torch/_refs/__init__.py:4690]: Not sure if x should be tensor or shape, as ```aten::uniform()``` expects tensor as the first argument:
```
>>> import torch
>>> torch.ops.aten.uniform([2, 3])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/scratch/ybliang/work/repos/pytorch/torch/_ops.py", line 500, in __call__
    return self._op(*args, **kwargs or {})
RuntimeError: Overloaded torch operator invoked from Python failed to many any schema:
aten::uniform() Expected a value of type 'Tensor' for argument 'self' but instead found type 'list'.
Position: 0
Value: [2, 3]
Declaration: aten::uniform(Tensor self, float from=0., float to=1., *, Generator? generator=None) -> Tensor
Cast error details: Unable to cast [2, 3] to Tensor

aten::uniform() Expected a value of type 'Tensor' for argument 'self' but instead found type 'list'.
Position: 0
Value: [2, 3]
Declaration: aten::uniform.out(Tensor self, float from=0., float to=1., *, Generator? generator=None, Tensor(a!) out) -> Tensor(a!)
Cast error details: Unable to cast [2, 3] to Tensor
```

Comment (yanboliang) [reply torch/_refs/__init__.py:4690]: And dispatch is at https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L7734

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -1929,6 +1930,21 @@ def norm(
     ).to(result_dtype)
 
 
+@register_decomposition(aten.uniform)
+def uniform(
+    x: Tensor,
+    low: Union[bool, int, float] = 0.0,
+    high: Union[bool, int, float] = 1.0,
+):
+    return prims.uniform(
```
Comment (mruberry) [inline torch/_decomp/decompositions.py:1939]: This might want to call `_uniform_helper` for consistency

File: test/test_decomp.py [hunk 3]
```diff
@@ -392,6 +392,19 @@ def test_quick(self, device, dtype, op):
...
+        dtype = torch.float32
+        x = torch.rand(size, dtype=dtype, device=device)
+        low = 0.3
+        high = 0.9
+
+        torch.manual_seed(123)
+        ref = torch.ops.aten.uniform(x, low, high)
+        torch.manual_seed(123)
+        res = torch._decomp.decompositions.uniform(x, low=low, high=high)
+        self.assertTrue(torch.allclose(ref, res))
```
Comment (mruberry) [inline test/test_decomp.py:406]: Use the approved testing functions, like `torch.testing.assert_close` or the test suite's override of unittest's `assertEqual` (by calling `self.assertEqual`)

File: test/test_decomp.py [hunk 4]
```diff
@@ -392,6 +392,19 @@ def test_quick(self, device, dtype, op):
     def test_comprehensive(self, device, dtype, op):
         self.do_cross_ref(device, dtype, op, run_all=True)
 
+    def test_uniform(self, device):
+        size = (2, 3, 4, 5)
+        dtype = torch.float32
+        x = torch.rand(size, dtype=dtype, device=device)
```
Comment (mruberry) [inline test/test_decomp.py:398]: Prefer `torch.testing.make_tensor` when writing tests

**Comment by mruberry** [comment]
Looks pretty reasonable but see some inline comments. Let's let @ngimel take a another look

**Review by ngimel — APPROVED** [review]
Thanks!

**Pull request closed without merging**